### PR TITLE
Adding a multiaddress key for SSB

### PR DESCRIPTION
**Pull request by gpicron**
This is to allow representing Secure Scuttlebutt addresses in MultiAddress format

### CODE REVIEW

**Comment by rvagg** [comment]
will defer to libp2p team on this, it seems reasonable to me although it appears to be among a bundle of protocols rather than handshakes; maybe that distinction is irrelevant though?

**Comment by gpicron** [comment]
Actually, SHS is very similar to noise.  In the spec, it is described as in 2 parts:

- https://ssbc.github.io/scuttlebutt-protocol-guide/#handshake : describes the handshake
- https://ssbc.github.io/scuttlebutt-protocol-guide/#box-stream: describe the encrypted stream encapsulation using the shared secrets (keys and nounces) produced by the handshake.

Historically, it was created before Noise was published.

**Comment by marten-seemann** [comment]
Tagging @dhuseby, since we recently talked about Secure Scuttlebutt.

File: table.csv
Comment (p-shahi) [inline table.csv:126]:
```suggestion
shs,                            multiaddr,      0x01c8,         draft,      Secure Scuttlebut - Secret Handshake Stream
```

The description in this file helps but would you be open to making the name itself more descriptive?

Comment (SgtPooki) [reply table.csv:126]: `Scuttlebut` -> `Scuttlebutt`

**Comment by dhuseby** [comment]
> So the shs handshake is similar to XKpsk0 Noise handshake but it is distinct...

What makes it distinct is the addition of the fixed network identifier key for the HMAC'ing of the initial initiator and responder hello messages ([https://ssbc.github.io/scuttlebutt-protocol-guide/#client-hello](https://ssbc.github.io/scuttlebutt-protocol-guide/#client-hello)).

Noise ([https://noiseprotocol.org/noise.html#protocol-names-and-modifiers](https://noiseprotocol.org/noise.html#protocol-names-and-modifiers)) and its extensions like Disco ([https://discocrypto.com/disco.html#protocol-naming](https://discocrypto.com/disco.html#protocol-naming)) mix the protocol name into the internal handshake state.

**Comment by gpicron** [comment]
And by the way, multi-address is supposed to be only loosely coupled to libp2p.  SSB has its own "multiaddress" and it would be good to progressively move SSB to less proprietary standards (even if they were created after SSB)...

**Comment by BigLep** [comment]
@p-shahi @dhuseby : what's the decision here?  Do we have an approve and merge?

**Comment by dhuseby** [comment]
@BigLep it looks to me like it just needs reviewer approval. I was weighing in on the fact that this is a distinct protocol and should have it's own line and identifier.

@vmx, @rvagg, @marten-seemann, @mxinden any reason not to merge this? If not, then approve it and merge it.

File: table.csv
Comment (p-shahi) [inline table.csv:127]:
```suggestion
shs,                            multiaddr,      0x01c8,         draft,      Secure Scuttlebutt - Secret Handshake Stream
```

**Review by mxinden — APPROVED** [review]
No objections from my end. Just needs a pull request on https://github.com/multiformats/multiaddr/ as well.

**Comment by rvagg** [comment]
can be merged when that typo is fixed

**Review by rvagg — APPROVED** [review]

**Pull request merged**
